Migrations/Worker.cs

```diff
@@ -0,0 +1,84 @@
+using System.Diagnostics;
+using DbUp.Reboot;
+using DbUp.Reboot.Engine;
+using DbUp.Reboot.ScriptProviders;
+using DbUp.Reboot.Support;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.Hosting;
+using Microsoft.Extensions.Logging;
+
+namespace IdentityServerConfig.Migrations;
+
+public class Worker : BackgroundService
+{
+    private readonly IHostApplicationLifetime _hostApplicationLifetime;
+    private readonly ILogger<Worker> _logger;
+    private readonly IConfiguration _configuration;
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="Worker" /> class.
+    /// </summary>
+    public Worker(ILogger<Worker> logger, IHostApplicationLifetime hostApplicationLifetime, IConfiguration configuration)
+    {
+        _logger = logger;
+        _hostApplicationLifetime = hostApplicationLifetime;
+        _configuration = configuration;
+    }
+
+    /// <inheritdoc />
+    protected override async Task ExecuteAsync(CancellationToken stoppingToken)
+    {
+        try
+        {
+            _logger.LogInformation("Starting migration");
+
+            await Task.Run(() =>
+            {
+                var connectionString = _configuration.GetConnectionString("AuditLog");
+                if (connectionString == null)
+                {
+                    throw new ArgumentException("ConnectionString cannot be null");
+                }
+                _logger.LogInformation("Connection string is {ConnectionString}", connectionString);
+
+                EnsureDatabase.For.SqlDatabase(connectionString);
+
+                var path = Path.GetDirectoryName(Process.GetCurrentProcess()?.MainModule?.FileName);
+                _logger.LogInformation(path);
+                var upgradeEngineBuilder = DeployChanges.To
+                    .SqlDatabase(connectionString, null)
+                    .WithScriptsFromFileSystem(path,
+                        new FileSystemScriptOptions { IncludeSubDirectories = true, Filter = (x) => x.Contains("DeploymentScripts") },
+                        new SqlScriptOptions { ScriptType = ScriptType.RunOnce, RunGroupOrder = 1 })
+                    .WithTransaction()
+                    .LogToConsole();
+
+                Task.Delay(10000);
```

Comment:
Why are we introducing this delay here? Is there a specific reason for that?

Reply:
I don't know why it was there (it was also included in the code that i referenced) but i seems unnecessary. But fixed in 5a020fb